forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/gcp/GCPBucketStorageAccessor.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.common.gcp;
+
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.springframework.beans.factory.annotation.Value;
+
+public class GCPBucketStorageAccessor {
+
+    private static String PROJECT_ID;
+
+    private static volatile Storage storage = null;
```

Comment:
Why is this volatile? Are you expecting multiple threads to access it?

Reply:
It's a singleton pattern with lazy initialisation and "thread safe without mutual exclusion" to avoid affect the performance using a synchronized method, multiple threads can access to it and run it in parallel and we can ensure the visibility property. 
We not need a one instance of Storage for each thread (rest call).

Reply:
But where is `storage` being set in this class?

Reply:
It was a bug in the source, fixed.